src/Search.php

```diff
@@ -1127,8 +1143,27 @@ public static function exclude_unwanted_attachments( $args ) {
 			$wpdb->prepare( $sql, $params->get_values() ) // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
 		);
 
-		$args['post__not_in'] = $unwanted_attachment_ids;
+		return $unwanted_attachment_ids;
+	}
 
-		return $args;
+	/**
+	 * Fetch all password protected posts that we don't want to include in search.
+	 *
+	 * @return array The password protected post ids.
+	 */
+	public static function get_password_protected_post_ids():array {
```

Comment:
For attachments we needed to execute another query, because wp_query didn't have a way to specify the post_mime_type in the args, so we had to add the first query. However for post password we probably don't need to do that and can add it directly to the query.
```
public static function exclude_unwanted_posts( $args ) {
   $args['has_password'] => false,
}
```
see https://stackoverflow.com/questions/7538959/how-to-exclude-password-protected-posts-in-wordpress-loop